src/ServerRequestFactoryInterface.php

```diff
@@ -12,20 +12,9 @@
      *
      * @param string $method
      * @param UriInterface|string $uri
+     * @param array $server optional server params ($_SERVER or similar structure)
      *
      * @return ServerRequestInterface
      */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from server variables.
-     *
-     * @param array $server Typically $_SERVER or similar structure.
-     *
-     * @return ServerRequestInterface
-     *
-     * @throws \InvalidArgumentException
-     *  If no valid method or URI can be determined.
-     */
-    public function createServerRequestFromArray(array $server);
+    public function createServerRequest($method, $uri, $server = []);
```

Comment:
From the PSR-7 docs:

> *typically* derived from PHP's `$_SERVER` superglobal.

> The data IS NOT REQUIRED to originate from `$_SERVER`

> the query params might not be in sync with the URI or server params

And so on.

Fairly certain we don't need to concern ourselves with this at all.